File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    int k_size;
    priority_queue<int,vector<int> , greater<int>> pq;
    KthLargest(int k, vector<int>& nums) {
        k_size = k;

        for(auto it : nums)
        {

            pq.push(it);
            if(pq.size()>k_size)
            {
                pq.pop();
            }
        }
        
    }
    
    int add(int val) {

        pq.push(val);
        if(pq.size()>k_size)
        {
            pq.pop();
        }

        return pq.top();
        
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.java
class KthLargest {
    PriorityQueue<Integer> pq;
    int k_size;

    public KthLargest(int k, int[] nums) {
        k_size = k;
        pq = new PriorityQueue<>(k);

        for(int n : nums)
        {
            pq.offer(n);
            if(pq.size()>k_size)
            {
                pq.poll();
            }
        }
    }
    
    public int add(int val) {
         
            pq.offer(val);
            if(pq.size()>k_size)
            {
                pq.poll();
            }
        return pq.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */